src/datatypes/datetime.ts

```diff
@@ -29,6 +29,14 @@ import {
 
 export const DATE_PRIVATE = Symbol.for("edgedb.datetime");
 
+function toNumber(val: any): number {
+  const n = Number(val);
+  if (Number.isNaN(n)) {
```

Comment:
Can you add a comment here? I'm not sure this is correct just by looking at it.

Reply:
Done